backend/canvas_app_explorer/canvas_lti_manager/django_factory.py

```diff
@@ -0,0 +1,18 @@
+from django.http import HttpRequest
+from canvas_oauth.oauth import get_oauth_token
+
+from .manager import CanvasLtiManager
+
+class DjangoCourseLtiManagerFactory:
+    """
+    Factory class creating CourseLtiManager instances using HttpRequest objects.
+    Assumes course_id is set in the session and that the Django Canvas OAuth app is set up.
+    """
+
+    def __init__(self, api_url: str):
+        self.api_url = api_url
+
+    def create_manager(self, request: HttpRequest) -> CanvasLtiManager:
```

Comment:
In the cases we're using this, we're actually passing a DRF `Request`, but my understanding is that it's an extension of `django.HttpRequest` (interface should be the same?). https://www.django-rest-framework.org/api-guide/requests/ Seemed better to use the more basic version in case this was use outside of a DRF layer.